demo_nodes_cpp/src/topics/listener_loaned_message.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_components/register_node_macro.hpp"
+
+#include "std_msgs/msg/float64.hpp"
+#include "demo_nodes_cpp/visibility_control.h"
+
+namespace demo_nodes_cpp
+{
+// Create a Listener class that subclasses the generic rclcpp::Node base class.
+// The main function below will instantiate the class as a ROS node.
+class LoanedMessageListener : public rclcpp::Node
```

Comment:
Personally, I would rename this class.  The fact that it is a loaned message under the hood isn't super-interesting to the user.  What's more interesting is that it is zero-copy, so I'd rename this to `ZeroCopyListener` or whatever.

Reply:
I kind of disagree with this. We have the equivalent talker which is called `LoanedMessageTalker` and I believe this pattern makes sense, nonetheless for consistency - we also have a pair of `SerializedMessage[Talker|Listener]`.

The idiom of a loaned message is needed for zero-copy, but you can perfectly fine use a loaned message without the middleware supporting zero-copy.

Reply:
> I kind of disagree with this. We have the equivalent talker which is called `LoanedMessageTalker` and I believe this pattern makes sense, nonetheless for consistency - we also have a pair of `SerializedMessage[Talker|Listener]`.
> 
> The idiom of a loaned message is needed for zero-copy, but you can perfectly fine use a loaned message without the middleware supporting zero-copy.

OK, that makes sense.  I think the thing I'm stuck on here is how the user knows that loaning is happening, and what they should expect different when loaning is used.